### PR TITLE
[feat]: [CI-8188]: 0777 permission must be given to /tmp/engine

### PR DESCRIPTION
**Pull request by anurag-harness**
Currently even though we create the dir with 0777, the default umask of 022 creates the dir with 0755. Hence we need to explicitly chmod to achieve this. Please refer to https://harness.atlassian.net/browse/CI-8107 for more details as to why its needed.

### CODE REVIEW

File: engine/engine.go [hunk 1]
```diff
@@ -52,19 +53,22 @@ func (e *Engine) Setup(ctx context.Context, pipelineConfig *spec.PipelineConfig)
 	}
 	// create volumes
 	for _, vol := range pipelineConfig.Volumes {
-		if vol != nil && vol.HostPath != nil {
-			path := vol.HostPath.Path
-			vol.HostPath.Path = pathConverter(path)
+		if vol == nil || vol.HostPath == nil {
```
Comment (anurag-harness) [inline engine/engine.go:56]: Had to make this change due to failed linter

https://harness.drone.io/harness/lite-engine/930/1/3

File: engine/engine.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ const (
 	DockerSockVolName  = "_docker"
 	DockerSockUnixPath = "/var/run/docker.sock"
 	DockerSockWinPath  = `\\.\pipe\docker_engine`
+	permissions        = 0777
```
Comment (hemanthmantri) [inline engine/engine.go:28]: s/permissions/Permissions

File: engine/engine.go [hunk 3]
```diff
@@ -52,19 +53,22 @@ func (e *Engine) Setup(ctx context.Context, pipelineConfig *spec.PipelineConfig)
...
+			continue
+		}
+		path := vol.HostPath.Path
+		vol.HostPath.Path = pathConverter(path)
 
-			if _, err := os.Stat(path); err == nil {
-				continue
-			}
+		if _, err := os.Stat(path); err == nil {
+			_ = os.Chmod(path, permissions)
```
Comment (hemanthmantri) [inline engine/engine.go:63]: Please add a log.

File: engine/engine.go [hunk 4]
```diff
@@ -52,19 +53,22 @@ func (e *Engine) Setup(ctx context.Context, pipelineConfig *spec.PipelineConfig)
...
 
-			if err := os.MkdirAll(path, 0777); err != nil { //nolint:gomnd
-				return errors.Wrap(err,
-					fmt.Sprintf("failed to create directory for host volume path: %q", path))
-			}
+		if err := os.MkdirAll(path, permissions); err != nil {
+			return errors.Wrap(err,
+				fmt.Sprintf("failed to create directory for host volume path: %q", path))
 		}
+		_ = os.Chmod(path, permissions)
```
Comment (hemanthmantri) [inline engine/engine.go:71]: Why is this needed?

**Review by harshjain12 — APPROVED** [review]

File: engine/engine.go (hunk 3 above)
Comment (anurag-harness) [reply engine/engine.go:63]: There is no logger here

File: engine/engine.go (hunk 4 above)
Comment (anurag-harness) [reply engine/engine.go:71]: We still need to give it 0777 if it was created right now
